### PR TITLE
Fix filtering, detecting, reseting and ranking closes #135

### LINKED ISSUE #135: Filtering, detecting, ranking is not working

### PR DESCRIPTION
**Pull request by Shine226**

### CODE REVIEW

File: wiggum_app/controller.py [hunk 1]
```diff
@@ -235,22 +234,35 @@ def main():
...
-            result_dict_dict = models.getResultDict(labeled_df_setup, filter_result, filter_object['subgroup'])
+            #return jsonify(result_dict_dict)
+            return jsonify(distance_heatmap_dict = distance_heatmap_dict, 
+                            result_df = filter_result.to_json(orient='records'),
+                            df = df)
 
-            return jsonify(result_dict_dict)
 
         # visualize.html 'Reset' button clicked
         if action == 'reset':
```
Comment (brownsarahm) [inline wiggum_app/controller.py]: Can you add a few lines of comments at each of these if cases that outlines what is done?

File: wiggum_app/controller.py [hunk 2]
```diff
@@ -286,10 +303,15 @@ def main():
...
-            result_dict_dict = {}
-            result_dict_dict = models.getResultDict(labeled_df_setup, rank_result)
+            # Generate distance heatmaps
+            distance_heatmap_dict = models.getDistanceHeatmapDict(rank_result)
+
+            df = labeled_df_setup.df.to_dict(orient='records')
+            df = json.dumps(df, indent=2)
 
-            return jsonify(result_dict_dict)
+            return jsonify(distance_heatmap_dict = distance_heatmap_dict,
```
Comment (brownsarahm) [inline wiggum_app/controller.py]: if filters have been applied they should still be applied after ranking. so maybe re-filter the list? after ranking if the table is filtered when the button is pushed?

Comment (Shine226) [reply wiggum_app/controller.py]: If filter_result can be stored in LabeledDataFrame, I can use it for ranking. If no filter, filter_result is equal to result_df. Or I can do filter process again before ranking in controller.py. Which one is good for now?

Comment (Shine226) [reply wiggum_app/controller.py]: I just tested on iris test dataset, click detect and then click rank button, didn't see any error. Can you give more detail about this error?

Comment (Shine226) [reply wiggum_app/controller.py]: If click "command" key and click the mouse at the same time in Mac, it can unselect the previous selected option.

**Comment by brownsarahm** [comment]
Ranking issues:

1. I can't rank until after detecting, which I don't think should be the case.
2. I set it to sum, distance rank and it added a mean view distance column.
3. when I change to max distance rank nothing happens. 
4. after a few sucessful ranking and filter then i tried sum distance rank again and got the error below

```
raceback (most recent call last):
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/flask/app.py", line 2463, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/flask/app.py", line 2449, in wsgi_app
    response = self.handle_exception(e)
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/flask/app.py", line 1866, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/flask/app.py", line 2446, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/flask/app.py", line 1951, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/flask/app.py", line 1820, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/flask/app.py", line 1949, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/flask/app.py", line 1935, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/wiggum_app/controller.py", line 313, in main
    result_df = rank_result.to_json(orient='records'),
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/pandas/core/generic.py", line 2425, in to_json
    index=index,
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/pandas/io/json/_json.py", line 78, in to_json
    index=index,
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/pandas/io/json/_json.py", line 122, in __init__
    self._format_axes()
  File "/home/smb/Documents/code/detect_simpsons_paradox_dev/env/lib/python3.6/site-packages/pandas/io/json/_json.py", line 211, in _format_axes
    "'{orient}'.".format(orient=self.orient)
ValueError: DataFrame columns must be unique for orient='records'.

```

detect issue:
if threshold is too high that nothing is returned, everything goes blank, but no error announce. please add a popup.

thanks for tip on how to unselect. we should note that somewhere. Some select feilds work by clicking again. 

FIlter issue:
1. reset should unselect the menu selections
2. if any filter settings are applied, they should be re-applied after detection. so, make a variable that's a part of wiggum app that's `current_filters` or something like that and hold those parameters there **or**  have a binary variable `filter` that is false default, switches to true when filter btton pressed and to false when reset is pressed.  or something similar.

**Pull request merged**